src/unix/stream.c

```diff
@@ -973,8 +973,20 @@ uv_handle_type uv__handle_type(int fd) {
   memset(&ss, 0, sizeof(ss));
   sslen = sizeof(ss);
 
-  if (getsockname(fd, (struct sockaddr*)&ss, &sslen))
+  if (getsockname(fd, (struct sockaddr*)&ss, &sslen)) {
+#if defined(_AIX)
+    // On aix, ibmi, receiving RST from TCP instead of FIN immediately puts
+    // fd into an error state.
+    // In such case getsockname will return EINVAL, even if sockaddr_storage is
+    // valid. We ignore this and return UV_TCP to receive the last readable
+    // bytes
```

Comment:
I think this sentence also violates the TCP spec. It is not very clear on this point (https://datatracker.ietf.org/doc/html/rfc793), but when RST is received, the receiver is generally expected to process it immediately (out of order), then to delete the TCB, destroying all read and write buffers in the process.
```
        If the RST bit is set then, any outstanding RECEIVEs and SEND
        should receive "reset" responses.  All segment queues should be
        flushed.  Users should also receive an unsolicited general
        "connection reset" signal.  Enter the CLOSED state, delete the
        TCB, and return.
```
```suggestion
    // valid. In such cases, we will permit the user to open the connection as uv_tcp still, so that the user can get immediately notified of the error in their read callback and close this fd.
```